app/controllers/tweeets_controller.rb

```diff
@@ -0,0 +1,71 @@
+class TweeetsController < ApplicationController
+  before_action :set_tweeet, only: %i[show edit update destroy]
+  before_action :authenticate_user!, except: %i[index show]

```

Comment:
- **_There should be a `before_action` that restrict access to the `:new` and `:create` methods ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication)). This functionality might have been covered by `before_action :authenticate_user!` however the requirement specifically mentioned `before_action` filter for `new` and `create` methods_** because it will help you guys understand how to create your authentications